src/categorical_algebra/CSets.jl

```diff
@@ -1734,6 +1734,28 @@ function Base.iterate(Sub::SubobjectIterator, state=SubobjectIteratorState())
 end
 
 
+struct OverlapIterator
+  top::ACSet
+  others::Vector{ACSet}
+  function OverlapIterator(Xs::Vector{T}) where T<:ACSet
+    t, o... = Xs[sortperm(total_parts.(Xs))]
+    new(t, o)
+  end
+end
+
+function Base.collect(itr::OverlapIterator)
+  res = []; for sp in itr push!(res, sp) end; res
```

Comment:
Is this method needed? Given that the iteration interface is implemented, I would expect `Base.collect` to already do this.

Reply:
ah I just had to define Base.IteratorSize for OverlapIterator!